### PR TITLE
FIXED: Since XSD 1.1, value 0 is allowed for xsd:gYear.

### PR DESCRIPTION
**Pull request by wouterbeek**
Updated the tests accordingly.

### CODE REVIEW

File: test_xsd.pl [hunk 1]
```diff
@@ -60,28 +60,28 @@
 % Term --> String
 
 test(time_string, S == "2016-01-30") :-
-    xsd_time_string(date(2016,01,30), Y, S),
+    xsd_time_string(date(2016,1,30), Y, S),
```
Comment (JanWielemaker) [inline test_xsd.pl]: Why this change?  I wrote 01 instead of 1 for the visual compatibility with normal human ways to write down a date.  For Prolog `1` and `01` are the same.

File: xsd.c [hunk 2]
```diff
@@ -567,17 +559,6 @@ maybe_invalid_time_url(term_t type)
 }
 
 
-static int
```
Comment (JanWielemaker) [inline xsd.c]: I would be inclined to keep this function in there and simply change it to return `year*sign`.   The compiler will remove it anyway.  It makes the change easier to understand and would facilitate a mode to support the old XSD stuff if we would want to.

File: test_xsd.pl (hunk 1 above)
Comment (wouterbeek) [reply test_xsd.pl]: Yes you're right, that's just me being pedantic.  I've changed this back and made the use of a leading zero consistent throughout the test file.

File: xsd.c (hunk 2 above)
Comment (wouterbeek) [reply xsd.c]: Yes, that's a good idea.  I've now added XSD 1.0 behavior through a C preprocessor directive.  Is this indeed the best way to achieve this?

I would like to make the Prolog test file depend on this directive as well, running either XSD 1.0 or 1.1 tests, based on the version for which `xsd.c` was compiled.  Is there a way to communicate this preprocessor directive between the C source file and the Prolog test file?

**Comment by JanWielemaker** [comment]
Thanks.  Merged and pushed.  I squashed the two commits and edited the commit message.  You cannot get access to CPP flags.  The only thing you can do is add a predicate, say xsd_version/1 that returns the current version depending on the preprocessor flag.  If you intend to do so, I suggest to add xsd_version/2 that also allows setting the version (maintained in some global C variable).   Please use a new PR for that.
